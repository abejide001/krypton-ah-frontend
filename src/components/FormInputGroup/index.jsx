import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'semantic-ui-react';

function FormInput({
  type, placeholder, name, value, handleChange
}) {
  return (
    <Form.Input
      type={type}
      placeholder={placeholder}
      name={name}
      value={value}
      fluid
      required
      onChange={handleChange}
    />
  );
}

FormInput.propTypes = {
  type: PropTypes.string,
  placeholder: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.string,
  handleChange: null
};

FormInput.defaultProps = {
  type: 'text',
  placeholder: 'Email Address',
  name: 'text',
  value: 'email',
  handleChange: null
};

export default FormInput;
