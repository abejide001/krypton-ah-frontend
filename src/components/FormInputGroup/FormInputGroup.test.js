import React from 'react';
import { shallow } from 'enzyme';
import FormInputGroup from '.';

describe('Modal FormInput', () => {
  it('it should matches the snapshot', () => {
    const wrapper = shallow(<FormInputGroup />);
    expect(wrapper).toMatchSnapshot();
  });
});
