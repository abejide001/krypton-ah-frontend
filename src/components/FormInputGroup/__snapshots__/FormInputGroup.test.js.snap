// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal FormInput it should matches the snapshot 1`] = `
<FormInput
  as={[Function]}
  control={[Function]}
  fluid={true}
  name="text"
  onChange={null}
  placeholder="Email Address"
  required={true}
  type="text"
  value="email"
/>
`;
